src/constraints.jl

```diff
@@ -513,33 +513,37 @@ function check_belongs_to_model(con::VectorConstraint, model)
 end
 
 function moi_add_constraint(
-    model::MOI.ModelLike,
-    f::MOI.AbstractFunction,
-    s::MOI.AbstractSet,
-)
-    if !MOI.supports_constraint(model, typeof(f), typeof(s))
-        if moi_mode(model) == DIRECT
-            bridge_message = "."
-        elseif moi_bridge_constraints(model)
-            error(
-                sprint(
-                    io -> MOI.Bridges.debug(
-                        model.optimizer,
-                        typeof(f),
-                        typeof(s);
-                        io = io,
-                    ),
-                ),
-            )
-        else
-            bridge_message = ", try using `bridge_constraints=true` in the `JuMP.Model` constructor if you believe the constraint can be reformulated to constraints supported by the solver."
-        end
+    model::JuMP.Model,
+    f::F,
+    s::S,
+) where {F<:MOI.AbstractFunction,S<:MOI.AbstractSet}
+    return moi_add_constraint(model, backend(model), f, s)
+end
+
+function moi_add_constraint(
+    model::JuMP.Model,
+    moi_backend::MOI.ModelLike,
+    f::F,
+    s::S,
+) where {F<:MOI.AbstractFunction,S<:MOI.AbstractSet}
+    if MOI.supports_constraint(moi_backend, F, S)
+        # Our backend supports the constraint. Go ahead and add it.
+        return MOI.add_constraint(moi_backend, f, s)
+    elseif _add_bridges_if_needed(model)
```

Comment:
To keep the same behavior as before, this should be
```julia
_add_bridges_if_needed(model) && MOI.supports_constraint(model, F, S)
```

Reply:
Nah this is a recursive call. It checks the `supports_constraint` on the next call.